src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
I disagree that where program logic is implemented is not relevant because I think this design incorrectly places responsibility of one class in another. I also don't agree that object use after disposal as a clear user error - it's an API usage scenario that needs to be considered carefully. 

As an application developer, I may need to track telemetry during application shutdown in code don't understand or control, such as in a legacy system or a system where object disposal is managed by a framework I don't understand. If I'm investigating a crash or a hang at shutdown of my application, I might want to explicitly flush the telemetry channel explicitly without knowing it was already disposed. I'd be frustrated when this attempt resulted in an `ObjectDisposedException` hiding my original problem and leaving the telemetry unsent. In this scenario, I'd want a forgiving API that helps me solve my problem instead of adding insult to injury by saying I flushed at wrong time.

Could you take a look at implementation of the `InMemoryTransmitter.Flush`? I may be mistaken, but it looks safe to invoke after `InMemoryTransmitter.Dispose`. If that's the case, does the `InMemoryChannel.Flush` really need to throw the `ObjectDisposedException`?